### PR TITLE
20744: Updates docs for react aggregate api

### PR DESCRIPTION
**Pull request by jackx111**

### CODE REVIEW

**Comment by jackx111** [comment]
Also made the name of trainee consistent, so t. becomes trainee.

File: source/user_guide/advanced_capabilities/case_importance.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ Concepts & Terminology
 
 How-To Guide
 ------------
-Case importance is similiar to feature importance in that it comprises of two metrics, case mean decrease in accuracy (mda) and case contribution. 
+Case importance is similiar to feature importance in that it comprises of two metrics, case mean decrease in accuracy (mda) and case contribution.
```
Comment (jdbeel) [inline source/user_guide/advanced_capabilities/case_importance.rst:32]:
```suggestion
Case importance is similar to feature importance in that it comprises of two metrics, case mean decrease in accuracy (MDA) and case contribution.
```

File: source/user_guide/advanced_capabilities/case_importance.rst [hunk 2]
```diff
@@ -106,22 +106,22 @@ The code from all of the steps in this guide is combined below:
     action_features = ['target']
     context_features = features.get_names(without=action_features)
 
-    trainee = Trainee(features=features)
+     trainee = Trainee(features=features)
```
Comment (jdbeel) [inline source/user_guide/advanced_capabilities/case_importance.rst:109]:
```suggestion
    trainee = Trainee(features=features)
```

File: source/user_guide/advanced_capabilities/feature_importance.rst [hunk 3]
```diff
@@ -51,53 +51,51 @@ The created :class:`~Trainee` will be referenced as ``trainee`` in the sections
...
-metrics may be specified together, but for this example they are separated. If robust metrics are calculated, then the ``robust`` parameter must be set to ``True`` to retrieve these metrics. If non-robust metrics are calculated, then the ``robust`` parameter can be set to the default value.
-
-.. code-block:: python
-
-    robust_feature_contributions = t.get_prediction_stats(action_feature=action_features[0], robust=True, stats=['contribution'])
-    robust_feature_mda = t.get_prediction_stats(action_feature=action_features[0], robust=True, stats=['mda'])
+    feature_mda_robust = trainee.react_aggregate(
+        context_features=context_features,
+        feature_influences_action_feature=action_features[0],
+        details={"feature_mda_robust" : True}
```
Comment (jdbeel) [inline source/user_guide/advanced_capabilities/feature_importance.rst:70]: The other edits in this PR use single quotes, I'm more partial to double quotes personally but we should at least keep it consistent.

```suggestion
        details={'feature_mda_robust' : True}
```

File: source/user_guide/basic_capabilities/predictions.rst [hunk 4]
```diff
@@ -243,17 +246,28 @@ Combined Code
...
    context_features = features.get_names(without=action_features)
 
-   t.analyze(context_features=context_features, action_features=action_features)
+   trainee.analyze(context_features=context_features, action_features=action_features)
+
+   # Recommended metrics
+   stats = trainee.react_aggregate(
+      action_feature=action_features[0],
+      details={
+         "prediction_stats": True,
```
Comment (jdbeel) [inline source/user_guide/basic_capabilities/predictions.rst:265]: Should keep quotes consistent:

```suggestion
         'prediction_stats': True,
```

File: source/user_guide/basic_capabilities/predictions.rst [hunk 5]
```diff
@@ -243,17 +246,28 @@ Combined Code
...
+         "selected_prediction_stats": ['rmse', 'spearman_coeff', 'r2', 'mae']
```
Comment (jdbeel) [inline source/user_guide/basic_capabilities/predictions.rst:266]: Should keep quotes consistent:

```suggestion
         'selected_prediction_stats': ['rmse', 'spearman_coeff', 'r2', 'mae']
```

File: source/user_guide/basic_capabilities/residuals.rst [hunk 6]
```diff
@@ -40,44 +40,37 @@ The created :class:`~Trainee` will be referenced as ``trainee`` in the sections
...
+Global metrics are retrieved through using :py:meth:`Trainee.react_aggregate`.  Both :ref:`robust` and non-robust (full) versions are also available.
 
 .. code-block:: python
 
-    t.react_into_trainee(
-        context_features=context_features,
+    residuals = trainee.react_react_aggregate(
         action_feature=action_features[0],
-        residuals_robust=True
+        details={"feature_residuals_full": True}
```
Comment (jdbeel) [inline source/user_guide/basic_capabilities/residuals.rst:72]: Should keep quotes consistent

```suggestion
        details={'feature_residuals_full': True}
```

File: source/user_guide/basic_capabilities/residuals.rst [hunk 7]
```diff
@@ -125,24 +115,20 @@ The code from all of the steps in this guide is combined below:
...
-    trainee.react_into_trainee(
-        context_features=context_features,
-        action_feature=action_features[0],
-        residuals_robust=True
-    )
-    residuals = trainee.get_prediction_stats(
+    residuals = trainee.react_react_aggregate(
         action_feature=action_features[0],
-        stats=['mae']
+        details={"feature_residuals_full": True}
```
Comment (jdbeel) [inline source/user_guide/basic_capabilities/residuals.rst:123]: Should keep quotes consistent

```suggestion
        details={'feature_residuals_full': True}
```

File: source/user_guide/concepts/global_vs_local.rst [hunk 8]
```diff
@@ -58,16 +59,16 @@ Local metrics are controlled through the ``details`` parameter in :py:meth:`Trai
 
 .. code-block:: python
 
-    details = {"feature_contributions" : True}
+    details = {"feature_contributions_robust" : True}
```
Comment (jdbeel) [inline source/user_guide/concepts/global_vs_local.rst:62]: Should keep quotes consistent

```suggestion
    details = {'feature_contributions_robust' : True}
```

File: source/user_guide/concepts/model_performance.rst [hunk 9]
```diff
@@ -54,27 +54,32 @@ performance stats.
 
 .. code-block:: python
 
-    # Recommended metrics
-    trainee.react_into_trainee(action_feature=action_features[0], residuals=True)
-    stats = trainee.get_prediction_stats()
-    accuracy = stats["accuracy"]
+    prediction_stats = trainee.react_react_aggregate(
+        details={"prediction_stats": True}
```
Comment (jdbeel) [inline source/user_guide/concepts/model_performance.rst:58]: Should keep quotes consistent

```suggestion
        details={'prediction_stats': True}
```

File: source/user_guide/concepts/model_performance.rst [hunk 10]
```diff
@@ -54,27 +54,32 @@ performance stats.
...
 The user from the example would want to do the following:
 
 .. code-block:: python
 
     # Conditions for continuous features take a tuple
     # that represents the range of the condition.
-    performance_stats = trainee.get_prediction_stats(condition={"age": [40, 9999]})
+    performance_stats = trainee.react_react_aggregate(
+        details={
+            "prediction_stats": True,
```
Comment (jdbeel) [inline source/user_guide/concepts/model_performance.rst:78]: Should keep quotes consistent

```suggestion
            'prediction_stats': True,
```

File: source/user_guide/concepts/model_performance.rst [hunk 11]
```diff
@@ -54,27 +54,32 @@ performance stats.
...
+            "action_condition": {"age": [40, 9999]}
```
Comment (jdbeel) [inline source/user_guide/concepts/model_performance.rst:79]: Should keep quotes consistent

```suggestion
            'action_condition': {'age': [40, 9999]}
```

File: source/user_guide/concepts/model_performance.rst [hunk 12]
```diff
@@ -84,11 +89,13 @@ on cases that represent individuals over the age of 40, who are also married and
 
     # Conditions for continuous features take a tuple
     # that represents the range of the condition.
-    performance_stats = trainee.get_prediction_stats(
-        condition={
-            "age": [40, 9999],
-            "marital-status": "married",
-            "job-status": "unemployed",
+    performance_stats = trainee.react_react_aggregate(
+        details={
```
Comment (jdbeel) [inline source/user_guide/concepts/model_performance.rst:93]: Should keep these quotes consistent

**Review by jdbeel — CHANGES_REQUESTED** [review]

**Review by jdbeel — APPROVED** [review]

**Review by jdbeel — APPROVED** [review]

**Pull request merged**